PlaidLink.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useEffect } from 'react';
+import {
+  Linking,
+  NativeEventEmitter,
+  NativeModules,
+  Platform,
+  Pressable,
+} from 'react-native';
+import {
+  LinkError,
+  LinkEventListener,
+  LinkExit,
+  LinkSuccess,
+  PlaidLinkProps,
+} from './types/Types';
+
+/**
+ * A hook that registers a listener on the plaid emitter for the 'onEvent' type.
+ * The listener is cleaned up when this view is unmounted
+ *
+ * @param LinkEventListener the listener to call
+ */
+export const usePlaidEmitter = (LinkEventListener: LinkEventListener) => {
+  useEffect(() => {
+    const emitter = new NativeEventEmitter(
+      Platform.OS === 'ios'
+        ? NativeModules.RNLinksdk
+        : NativeModules.PlaidAndroid,
+    );
+    const listener = emitter.addListener('onEvent', LinkEventListener);
+    // Clean up after this effect:
+    return function cleanup() {
+      listener.remove();
+    };
+  }, []);
+};
+
+
+export const openLink = async (plaidLinkProps: PlaidLinkProps) => {
+  if (Platform.OS === 'android') {
+    NativeModules.PlaidAndroid.startLinkActivityForResult(
+      JSON.stringify(plaidLinkProps.config),
+      (result: LinkSuccess) => {
+        if (plaidLinkProps.onSuccess != null) {
+            plaidLinkProps.onSuccess(result);
+        }
+      },
+      (result: LinkExit) => {
+        if (plaidLinkProps.onExit != null) {
+            plaidLinkProps.onExit(result);
+        }
+      },
+    );
+  } else {
+    NativeModules.RNLinksdk.create(plaidLinkProps.config);
+    NativeModules.RNLinksdk.open(
+      // TODO(what is the type of metadata here?)
```

Comment:
Not sure how smart typescript is, is there a way to refine a top-level type using a Utility type?

e.g.

type Container {
  let a: A
}

If you want the type of Container.a can you do `Pick<Container, A>`?

If so it would be `Pick<LinkSuccess, "metadata"> & Pick<LinkExit, "metadata">`.

I'm not sure if doing result.metadata is smart enough to generate the Pick implicitly

Reply:
I think I just need to remove this TODO. Since it's a map I made result a super set of all the props so I think that should work below?